### PR TITLE
Stellar port V2

### PR DESCRIPTION
**Pull request by llehmann-ledger**

### CODE REVIEW

File: ledger-core-stellar/src/stellar/stellarNetworks.cpp [hunk 1]
```diff
@@ -0,0 +1,58 @@
+/*
+ *
+ * tezosNetworks
```
Comment (gagbo) [inline ledger-core-stellar/src/stellar/stellarNetworks.cpp:3]: not Tezos :)

File: ledger-core-stellar/test/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    xdr_tests.cpp
+)
+
+target_link_libraries(ledger-core-stellar-tests ledger-core-bundle)
+target_link_libraries(ledger-core-stellar-tests ledger-test)
+target_link_libraries(ledger-core-stellar-tests ledger-core-integration-test)
+target_link_libraries(ledger-core-stellar-tests ledger-qt-host)
+target_link_libraries(ledger-core-stellar-tests ledger-core-stellar-obj)
+target_link_libraries(ledger-core-stellar-tests gtest gtest_main)
+
```
Comment (gagbo) [inline ledger-core-stellar/test/CMakeLists.txt:21]: This defines no test to run in `ctest`, can you please add some ?

File: ledger-core/inc/core/address/Address.hpp [hunk 3]
```diff
@@ -41,6 +41,7 @@ namespace ledger {
         public:
             explicit Address(const api::Currency& currency, const Option<std::string>& path);
             api::Currency getCurrency() override;
+            api::Currency getCurrency() const;
```
Comment (gagbo) [inline ledger-core/inc/core/address/Address.hpp:44]: You should not need to modify `ledger-core` directory for integrations. Why did you need it ? Maybe you can either directly use the private `api::Currency` member, or we should change the getter to be const from the start

File: ledger-core/inc/core/events/LambdaEventReceiver.hpp [hunk 4]
```diff
@@ -45,5 +45,18 @@ namespace ledger {
...
+         * @param promise
+         * @param successCodes
+         * @param failureCodes
+         * @return
+         */
+        std::shared_ptr<LambdaEventReceiver> make_promise_receiver(
+                Promise<Unit>& promise,
+                const std::vector<api::EventCode> &successCodes,
+                const std::vector<api::EventCode> &failureCodes
+        );
```
Comment (gagbo) [inline ledger-core/inc/core/events/LambdaEventReceiver.hpp:60]: Self note ; same here : why is make_promise_receiver not already in ledger-core if it's needed ?

File: ledger-core/inc/core/math/BigInt.hpp [hunk 5]
```diff
@@ -217,7 +225,8 @@ namespace ledger {
 
             template <typename T>
             BigInt& assignScalar(T value) {
-                auto bytes = endianness::scalar_type_to_array<T>(std::abs(value), endianness::Endianness::BIG);
+                auto abs_val = (value<0) ? -value : value;
+                auto bytes = endianness::scalar_type_to_array<T>(value, endianness::Endianness::BIG);
```
Comment (gagbo) [inline ledger-core/inc/core/math/BigInt.hpp:229]: Why is this change necessary ? Also, can these be `const` ?

File: ledger-core/inc/core/utils/JsonParserPath.hpp [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+#define LEDGER_CORE_JSONPARSERPATH_HPP
+
+#include <string>
+#include <list>
+#include <boost/variant.hpp>
+
+namespace ledger {
+    namespace core {
+
+        enum class JsonParserPathNodeType {OBJECT, ARRAY, VALUE};
```
Comment (gagbo) [inline ledger-core/inc/core/utils/JsonParserPath.hpp:42]: Are there tests for this class ?

File: ledger-core/src/core/utils/URL.cpp [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+#include <core/utils/URL.h>
+#include <sstream>
+#include <iomanip>
+#include <cctype>
+#include <core/utils/Hex.hpp>
+
+namespace ledger {
+    namespace core {
+        namespace url {
+            std::string encodeUrlQuery(const std::string &urlQuery) {
```
Comment (gagbo) [inline ledger-core/src/core/utils/URL.cpp:41]: Same here, we probably want to add unit tests for this

**Comment by llehmann-ledger** [comment]
> Also, it'd be nice to know which commit from develop you used as a source for the migration, so we can easily track which fixes went into v1 before you started the migration and be able to track them to port later

Sure,  I believe it was commit `5efb46b052194daf412fd75808c703ac376479c4` from the `LedgerHQ` repository

File: .gitmodules [hunk 8]
```diff
@@ -28,3 +28,6 @@
 [submodule "msgpack-c"]
 	path = ledger-core-algorand/src/thirdparty/msgpack-c
 	url = https://github.com/ledgerHQ/msgpack-c
+[submodule "ledger-core/lib/CRCpp"]
```
[the remark this answers is not in the document]
Comment (llehmann-ledger) [reply .gitmodules:31]: I believe it's already the case as there is this line in `circleci/init_submodules.sh` : `git submodule update -- core/lib/CRCpp || echo "===========CRCpp submodule already updated"
` ?

[the remark this answers is not in the document]
Comment (gagbo) [reply .gitmodules:31]: oh ok, I don't know how the CI was able to pass before then

File: ledger-core/inc/core/utils/JsonParserPath.hpp (hunk 6 above)
Comment (llehmann-ledger) [reply ledger-core/inc/core/utils/JsonParserPath.hpp:42]: Yep, tests are in the `json_test.cpp` file which is now included in the last commit

File: ledger-core/inc/core/math/BigInt.hpp (hunk 5 above)
Comment (llehmann-ledger) [reply ledger-core/inc/core/math/BigInt.hpp:229]: It is necessary because Clang (and maybe also gcc) complains about `std::abs()` being `ambiguous` there

File: ledger-core-stellar/idl/idl.djinni [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+    # Get recommended fee
+    getFeeStats(callback: Callback<StellarLikeFeeStats>);
+
+    # Get signers for this account
+    getSigners(callback: ListCallback<StellarLikeAccountSigner>);
+}
+
+StellarLikeWallet = interface +c {
+    exists(address: string, callback: Callback<bool>);
+}
```
[the remark this answers is not in the document]
Comment (llehmann-ledger) [reply ledger-core-stellar/idl/idl.djinni:158]: Thanks, the missing functions are now implemented

File: ledger-core-stellar/src/CMakeLists.txt [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+# set(CMAKE_INCLUDE_CURRENT_DIR ON)
+# set(ANDROID_CPP_FEATURES exceptions)
+
+# Automatically add API files to the library
+file(GLOB API_HEADER_FILES "${PROJECT_SOURCE_DIR}/inc/stellar/api/*.hpp")
+
+# Configure JNI target
+if (TARGET_JNI)
+    file(GLOB JNI_HEADER_FILES "${PROJECT_SOURCE_DIR}/inc/stellar/jni/*.hpp")
+    file(GLOB JNI_HEADER_FILES "${PROJECT_SOURCE_DIR}/inc/bitcoin/jni/*.hpp")
```
[the remark this answers is not in the document]
Comment (llehmann-ledger) [reply ledger-core-stellar/src/CMakeLists.txt]: We don't, it was a mistake :)

File: ledger-core/inc/core/address/Address.hpp (hunk 3 above)
Comment (llehmann-ledger) [reply ledger-core/inc/core/address/Address.hpp]: Okay, I changed it to the latter

**Comment by llehmann-ledger** [comment]
Can I merge the PR ?

**Comment by llehmann-ledger** [comment]
New commit : `b3f548d7901a8f72c86c6373b26331e9eb1f058d`

File: ledger-core-stellar/src/CMakeLists.txt [hunk 11]
```diff
@@ -0,0 +1,37 @@
+cmake_minimum_required(VERSION 3.12)
+
+# set(CMAKE_INCLUDE_CURRENT_DIR ON)
+# set(ANDROID_CPP_FEATURES exceptions)
```
[the remark this answers is not in the document]
Comment (llehmann-ledger) [reply ledger-core-stellar/src/CMakeLists.txt:4]: Sure

File: ledger-core-stellar/inc/stellar/transaction_builders/StellarLikeTransaction.hpp [hunk 12]
```diff
@@ -57,7 +57,7 @@ namespace ledger {
 
             std::vector<uint8_t> toSignatureBase() override;
 
-            const stellar::xdr::TransactionEnvelope& envelope() const { return _envelope; }
+            const stellar::xdr::TransactionEnvelope& envelope() const { return _envelope; };
```
[the remark this answers is not in the document]
Comment (llehmann-ledger) [reply ledger-core-stellar/inc/stellar/transaction_builders/StellarLikeTransaction.hpp]: Done

File: ledger-core-stellar/inc/stellar/xdr/StellarModelUtils.hpp [hunk 13]
```diff
@@ -0,0 +1,45 @@
...
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+
+#ifndef LEDGER_CORE_STELLARMODELUTILS_HPP
```
[the remark this answers is not in the document]
Comment (llehmann-ledger) [reply ledger-core-stellar/inc/stellar/xdr/StellarModelUtils.hpp]: Done
